File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.java
import java.util.*;
class Solution {
    public int[][] solution(int[][] data, String ext, int val_ext, String sort_by) {
        Map<String,Integer> map = new HashMap<>();
            map.put("code",0);
            map.put("date",1);
            map.put("maximum",2);
            map.put("remain",3);
        
        List<int[]> list = new ArrayList<>();
        for(int i=0; i<data.length; i++){
            if(data[i][map.get(ext)] < val_ext){
                list.add(data[i]);
            }
        }
        list.sort((a,b)->a[map.get(sort_by)] - b[map.get(sort_by)]);
        
        int[][] answer = new int[list.size()][];
        for(int i=0; i<list.size(); i++){
            answer[i] = list.get(i);
        }
        return answer;
    }
}

File: 프로그래머스/1/250121. ［PCCE 기출문제］ 10번 ／ 데이터 분석/［PCCE 기출문제］ 10번 ／ 데이터 분석.py
def solution(data, ext, val_ext, sort_by):
    answer = []
    group = []
    for code,date,maximum,remain in data:
        group.append({
            "code":code,
            "date":date,
            "maximum":maximum,
            "remain":remain
        })
    for g in group:
        if(int(g[ext]) < int(val_ext)):
            answer.append([g["code"],g["date"],g["maximum"],g["remain"]])
    key_idx = {
            "code":0,
            "date":1,
            "maximum":2,
            "remain":3
        }
    sorted_answer = sorted(answer,key=lambda x: x[key_idx[sort_by]])    
    return sorted_answer